bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/registration/ServletRegistration.java

```diff
@@ -37,9 +37,11 @@ public class ServletRegistration extends EndpointRegistration<ServletDTO> {
 	static {
 		ServiceLoader<MultipartSupportFactory> loader = ServiceLoader.load(MultipartSupportFactory.class);
 
-		for (MultipartSupportFactory element : loader) {
+		Iterator<MultipartSupportFactory> iterator = loader.iterator();
+
+		while (iterator.hasNext()) {
 			try {
-				factory = element;
+				factory = iterator.next();
```

Comment:
Just wondering, why the h*** is this done at all in the **static** initlizer and not in the if(....) block some lines below??
```
if (servletDTO.multipartEnabled) {
 .... 
}
```

Reply:
I honestly have lot of other problems, like broken help etc :-)
I haven't seen this code before and hope not see it anymore in the future :-)

Reply:
> Just wondering, why the h*** is this done at all in the **static** initlizer and not in the if(....) block some lines below??
> 
> ```
> if (servletDTO.multipartEnabled) {
>  .... 
> }
> ```

I would guess to avoid calling the `ServiceLoader` each time.

Reply:
> I honestly have lot of other problems, like broken help etc :-) I haven't seen this code before and hope not see it anymore in the future :-)

Thanks for stepping up to clean up the mess cleaning up the code can cause.